### PR TITLE
Use new GAX in Bigtable and Spanner

### PR DESCRIPTION
**Pull request by jskeet**
There's one further change that will need a GAX beta02, annoyingly... but we can get started with this.

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public async Task ScheduledMaintenanceEvictsSessions()
             var sessionPool = new SessionPool(client, options, client.Logger);
             var acquisitionTask = sessionPool.AcquireSessionAsync(s_sampleDatabaseName, new TransactionOptions(), default);
 
-            await client.Scheduler.RunAndPause(TimeSpan.FromMinutes(1));
+            await client.Scheduler.RunAsync(TimeSpan.FromMinutes(1));
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.cs]: super nit: In in TargettedSessionPoolTests the 1 minute ones are `RunForSecondsAsync(60)`, maybe having them consistent to one or the other? Here and elsewhere.

Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.cs]: Will do for anything 1 minute or less, but keep the larger ones. I'll actually change this PR to include beta02 and the usage of Wait as well, so it'll be a little while before that's ready to review. Will ping you when it is.

**Comment by jskeet** [comment]
Okay, it's now ready for re-review. Changes:

- Use GAX 2.6.0-beta02 instead of GAX 2.6.0-beta01
- Use RunForSecondsAsync everywhere in TargetedSessionPoolTests
- Extra commit at the end to remove a TODO

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.cs (hunk 1 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.cs]: OK, that sounds good.

**Pull request merged**